src/StatusQ/Components/StatusListPicker.qml

```diff
@@ -197,40 +175,43 @@ Item {
 
         function getSelectedItemsText() {
             var res = ""
-            for(var i = 0; i < root.inputList.count; i++) {
-                var item = root.inputList.get(i)
-                if(item.selected) {
-                    if(res != "")
-                        res += ", "
-                    res += d.formatSymbolShortNameText(item.symbol, item.shortName)
-                }
+            for(var i = 0; i < selectedItems.count; i++) {
+                var item = selectedItems.get(i)
+                if(res != "")
+                    res += ", "
+                res += d.formatSymbolShortNameText(item.symbol, item.shortName)
             }
             return res
         }
+    }
 
+    width: 110
+    height: 38
+
+    Repeater {
         // Used to update the base input list model with the new selected property
-        function updateInputListModel(key, selected, isSingleSelection) {
-            // If it is a single selection we must ensure that when a new key is selected  others are cleared.
-            // It must be done manually because as there is the option of filterning, the model visualized changes
-            // and it could be possible to have a filtered list without an element selected (but in the base model
-            // it is) so, the binding to not selected will not be executed, as in the current filtered model
-            // there are no changes to update, just only the selected one.
-            for(var i = 0; i < root.inputList.count; i++) {
-                if(root.inputList.get(i).key === key) {
-                    root.inputList.get(i).selected = selected
-                }
-                else if(isSingleSelection && selected) {
-                     // Clear all the list
-                    root.inputList.get(i).selected = false
-                }
+        // If it is a single selection we must ensure that when a new key is selected others are cleared.
+        id: itemsSelectorHelper
+
+        function selectItem(modelData, checked) {
+            modelData.selected = checked
+            if (!root.multiSelection && checked) {
+                itemsSelectorHelper.uncheckOthersThan(modelData.key)
```

Comment:
I wonder why is this needed for? If you put all the checkboxes/radio buttons in a `ButtonGroup` (https://doc.qt.io/qt-5/qml-qtquick-controls2-buttongroup.html) then it should be done automatically

Reply:
Yeah, that's a good question. Unfortunately, the delegate of `StatusListPicker` is not `AbstractButton` based..